app/src/main/java/com/example/sharingang/items/FirestoreItemRepository.kt

```diff
@@ -0,0 +1,100 @@
+package com.example.sharingang.items
+
+import android.util.Log
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import com.example.sharingang.BuildConfig
+import com.google.firebase.firestore.Query
+import com.google.firebase.firestore.ktx.firestore
+import com.google.firebase.firestore.ktx.firestoreSettings
+import com.google.firebase.ktx.Firebase
+import kotlinx.coroutines.tasks.await
+import java.lang.Exception
+import javax.inject.Inject
+import javax.inject.Singleton
+
+private const val TAG = "FirestoreItemRepository"
+
+@Singleton
+/**
+ * Implementation of ItemRepository using the Firestore database
+ *
+ * During development it requires running the Firebase emulator (see README.md)
+ */
+class FirestoreItemRepository @Inject constructor() :
```

Comment:
Nitpick: I think the convention is to put doc comments *before* annotations, and we already did it that way in `ItemsViewModel`